### PR TITLE
[Fix] Update team members page permissions

### LINKED ISSUE #8232: 🐛 Pool operator unable to view fellow team members
## 🐛 Bug

Pool operator cannot view fellow team members through the teams table. 

## 📋 Steps to Reproduce

1. Log in as pool@test.com
2. Attempt to view team members for your teams, like Test Team
3. Observe error toast

## 📸 Screenshot

gif

## 🙋‍♀️ Proposed Solution

Guessing problem stems from the query in question for the page. 
Pool operators cannot utilize `users`

image

## ✅ Acceptance Criteria

- [ ] Pool operator abilities work as expected 
- [ ] Add members button only visible for platform admin and community managers

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #8232 

## 👋 Introduction

Removes actions/queries for roles that do not have access to them.

## 🕵️ Details

Only platform admins and community managers have the ability to modify teams members. This removes those actions from other roles to prevent unauthorized queries. This also has the benefit of deferring queries until the data is needed (not that it had a huge impact to begin with 😅 ).

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run dev`
2. Login as a platform admin
3. Navigate to `/admin/teams/{teamId}/members`
4. Confirm you can add, edit and delete memebers
5. Logout and log back in as community manager
6. Repeat steps 3-4
7. Logout and log back in as a request responder
8. Repeat steps 3-4 but confirm the actions are not available
9. Logout and log back in as a pool operator
10. Repeat steps 3-4 but confirm the actions are not available

## 📸 Screenshot

Screenshot 2023-12-21 120847

### CODE REVIEW

**Comment by petertgiles** [comment]
@JamesHuf could you please review this one?  It's pretty big but it would be great if you could at least run through the AC and testing instructions.

**Review by JamesHuf — COMMENTED** [review]
Everything seems to be working as I'd expect:
- platform admin and community manager can both view, add and remove team members
- request responder is unable to view the team at all
- pool operator can view but not add or remove team members

Editing a team members isn't really possible since there's currently only one role associated with teams.

**Review by JamesHuf — APPROVED** [review]
Looks good to me!

File: apps/web/src/pages/Teams/TeamMembersPage/components/useAvailableRoles.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      id
+      name
+      isTeamBased
+      displayName {
+        en
+        fr
+      }
+    }
+
+    userPublicProfiles {
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Teams/TeamMembersPage/components/useAvailableRoles.ts]: Good catch! Removed in [remove unused fields in roles query](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8890/commits/301262dcc84a2636b9a8043909eaaa30e47265ab)

File: apps/web/src/pages/Teams/TeamMembersPage/components/AddTeamMemberDialog.tsx [hunk 2]
```diff
@@ -109,7 +107,7 @@ AddTeamMemberDialogProps) => {
     description: "Label for the add member to team form",
   });
 
-  const fetchingUsers = availableUsers === null;
+  const fetchingUsers = users === null;
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/Teams/TeamMembersPage/components/AddTeamMemberDialog.tsx]: Oh yeah, looks like it is pretty much the same 😅 Removed in [remove redundant user fetch var](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8890/commits/8fc53a1ea6abb7179805303b98f7ea33a6816362)

**Pull request merged**